libs/core/src/lib/tabs/tab-list.component.ts

```diff
@@ -205,10 +205,12 @@ export class TabListComponent implements AfterContentInit, AfterViewInit, OnDest
 
     /** @hidden */
     _highlightActiveTab({ id }: HTMLElement): void {
-        const tab = this._tabArray.find((_tab) => _tab.id === id);
+        const tab = this._tabArray.find((_tab) => _tab.panel._panelId === id);
         if (tab) {
+            if (!tab.active) {
+                this.selectedTabChange.emit(tab.panel);
+            }
             this._activateStackedTab(tab.panel, false);
```

Comment:
```suggestion
            this._activateStackedTab(tab.panel, false);
            if (!tab.active) {
                this.selectedTabChange.emit(tab.panel);
            }
```

Reply:
Without this, it will just emit inactive tab and for some brief moment, tab-list user will consider emitted value as active, when in reality it is not active yet